qutebrowser/browser/network/networkmanager.py

```diff
@@ -241,12 +242,22 @@ def clear_rejected_ssl_errors(self, url):
     @pyqtSlot('QNetworkReply', 'QAuthenticator')
     def on_authentication_required(self, reply, authenticator):
         """Called when a website needs authentication."""
-        answer = self._ask("Username ({}):".format(authenticator.realm()),
-                           mode=usertypes.PromptMode.user_pwd,
-                           owner=reply)
-        if answer is not None:
-            authenticator.setUser(answer.user)
-            authenticator.setPassword(answer.password)
+        user, password = None, None
+        if not hasattr(reply, "_netrc_used"):
+            reply._netrc_used = True
+            net = netrc.netrc()
+            authenticators = net.authenticators(reply.url().host())
+            if authenticators:
+                (user, _, password) = authenticators
```

Comment:
The issue `pylint` reports here is because it doesn't do flow control yet, and thinks `authenticators` still can be `None` here - a `# pylint: disable=unpacking-non-sequence` should help.